src/test/java/rest/SearchResourceTest.java

```diff
@@ -274,6 +294,140 @@ public void testGetPersonByFullName()
+        expResult.add(new PersonDTO_OUT(person1)); //both from city1
+        expResult.add(new PersonDTO_OUT(person2)); //both from city1
+        String zip = city1.getZipCode(); //2800
+        String city = city1.getCity(); //Lyngby
+        List<PersonDTO_OUT> result;
+        
+        //Act
+         result = get("search/city?zip={zip}&city={city}", zip, city). then()
+                .assertThat()
+                .statusCode(HttpStatus.OK_200.getStatusCode())
+                .extract().body()
+                .jsonPath().getList(".", PersonDTO_OUT.class); //https://stackoverflow.com/a/53006523
+
+        //Assert
+        assertThat((result), equalTo(expResult));
+    }
+    
+    @Test
+    public void testGetPersonsByCity_Exception1(){
+        //Arrange
+        String zip = ""; //bad input
+        String city = ""; 
+        
+        given()
+                .get("search/city?zip={zip}&city={city}", zip, city).then()
+                .assertThat()
+                .statusCode(HttpStatus.BAD_REQUEST_400.getStatusCode()).
+                body("code", equalTo(400)).
+                body("message", equalTo("Missing Input"));
+    }
+    
+    @Test
+    public void testGetPersonsByCity_Exception2(){
+        //Arrange
+        String zip = "123"; //Incorrect input
+        String city = "NoCity"; 
+        
+        given()
+                .get("search/city?zip={zip}&city={city}", zip, city).then()
+                .assertThat()
+                .statusCode(HttpStatus.NOT_FOUND_404.getStatusCode()).
+                body("code", equalTo(404)).
+                body("message", equalTo("No Persons lives in that city."));
+    }
+    
+    @Test 
+    public void testGetAllZipCodes(){
+        //Arrange
+        List<Integer> expResult = new ArrayList();
+        expResult.add(Integer.valueOf(city1.getZipCode())); //2800
+        expResult.add(Integer.valueOf(city2.getZipCode())); //8000
+        expResult.add(Integer.valueOf(city3.getZipCode())); //9900
+        List<Integer> result;
+        
+        //Act
+        result = get("/search/zip/").then()
+                .assertThat()
+                .statusCode(HttpStatus.OK_200.getStatusCode())
+                .extract().body()
+                .jsonPath().getList("", int.class); //could probably be Integer.class but I'm scared to change anything now that it works.
+
+        //Assert
+////        System.out.println(result.getClass());
+////        System.out.println(expResult.getClass());
+////        System.out.println(expResult.get(0).getClass());
+////        System.out.println(result.get(0).getClass());
+////        System.out.println(result);
+////        System.out.println(expResult);
+        //assertThat((result), equalTo(expResult));
+        //assertEquals(expResult, result);
+        //assertThat(expResult, Matchers.containsInAnyOrder(result.get(0), result.get(1), result.get(2)));
+        assertThat("ARE THEY EQUAL", result, containsInAnyOrder(expResult.toArray()));
```

Comment:
Haha, what? "ARE THEY EQUAL"? What is going on here? How does this line work?

Reply:
assertThat(String**REASON**, Object**ToCompare**, containsInAnyOrder(Elements\*))

\* I use toArray() to list the all the elements instead of just the list (doesn't work.

Essentially it says

assertThat("message", {2800, 8000, 9000} , containsInAnyOrder([2800, 8000, 9000]\*))

\* Where the iterator (indexPosition) does not matter

Documentation:  http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html#containsInAnyOrder(java.util.Collection)

Reply:
Thanks man. Nice. Where does that message go?

Reply:
It's primarily for logging I believe

Reply:
Ahh alright.

Reply:
Annoying that you had to do it this way but nice solution